### PR TITLE
Implement SDK function to verify the signature

### PR DESCRIPTION
**Pull request by MichaelKim20**
I implemented the SDK function to verify the signature

Related to bpfkorea/stoa#30

### CODE REVIEW

**Comment by Geod24** [comment]
Why are there two PR (this and #6) with the same name ? Was it a mistake ?

**Comment by TrustHenry** [comment]
It still seems to need a Commit theorem.

**Comment by MichaelKim20** [comment]
> Why are there two PR (this and #6) with the same name ? Was it a mistake ?

#7 added two commites after #6.

**Comment by MichaelKim20** [comment]
I removed the dependency with the previous PR on this.
This PR had 4 commit.

File: tests/BASE32.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe ('BASE32', () =>
+{
+    it ('It decodes the PublicKey as BASE32 and then re-encodes it.', () =>
+    {
+        let address = "GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW";
+        let decode = base32Decode(address);
+        let endcode = base32Encode(Buffer.from(decode));
+        assert.equal(address, endcode.toString());
+    });
+});
```
Comment (TrustHenry) [inline tests/BASE32.test.ts:26]:
```suggestion
});

```

File: tests/BASE32.test.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+describe ('BASE32', () =>
+{
+    it ('It decodes the PublicKey as BASE32 and then re-encodes it.', () =>
+    {
+        let address = "GDD5RFGBIUAFCOXQA246BOUPHCK7ZL2NSHDU7DVAPNPTJJKVPJMNLQFW";
+        let decode = base32Decode(address);
+        let encode = base32Encode(Buffer.from(decode));
+        assert.strictEqual(address, encode.toString());
+    });
+});
```
Comment (Geod24) [inline tests/BASE32.test.ts:26]: Why this? It's simply testing another package.

File: src/modules/net/BOAClient.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     * @param signature {Signature} The signature of `message`.
+     * @param message {Buffer} The signed message. Should not include the signature.
+     * @param public_key {PublicKey} Instance of PublicKey
+     * @returns {boolean} `true` if the signature is valid
+     */
+    public verify (signature: Signature, message: Buffer, public_key: PublicKey): boolean
+    {
+        return public_key.verify(signature, message);
+    }
+}
```
Comment (Geod24) [inline src/modules/net/BOAClient.ts:39]: This doesn't seem useful at all. It's simply repeating what `public_key` does. Likewise, the tests are just copy-pasted. Why this?

**Comment by AndrejMitrovic** [comment]
> Two commits look superfluous, the rest LGTM.

Same comment with me.

Also I will suggest to follow our guidelines of introducing tests in the same commit which introduces the functionality, not into a separate commit.

File: src/modules/net/BOAClient.ts (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/net/BOAClient.ts:39]: That's a good idea.
We can provide functions related to the public key separately from BOACliet.

File: tests/BASE32.test.ts (hunk 2 above)
Comment (MichaelKim20) [reply tests/BASE32.test.ts:26]: I will delete the associated commit.
But the test was meaningful.

**Comment by MichaelKim20** [comment]
Ready for review

**Pull request merged**